docs/tutorials/go/build-an-ecommerce-app/index.md

```diff
@@ -13,14 +13,690 @@ last_update:
+You can also use it to boostrap some aspects of each session, by running Signals or Queries upon creation.
+Create a Starter as `start/main.go`:
+
+```go
+package main
+
+import (
+    "context"
+    "fmt"
+    "log"
+    "time"
+
+    "temporal-ecommerce/app"
+
+    "go.temporal.io/sdk/client"
+)
+
+func main() {
+    c, err := client.NewClient(client.Options{})
+    if err != nil {
+        log.Fatalln("unable to create Temporal client", err)
+    }
+    defer c.Close()
+
+    workflowID := "CART-" + fmt.Sprintf("%d", time.Now().Unix())
+
+    options := client.StartWorkflowOptions{
+        ID:        workflowID,
+        TaskQueue: "CART_TASK_QUEUE",
+    }
+
+    state := app.CartState{Items: make([]app.CartItem, 0)}
+    we, err := c.ExecuteWorkflow(context.Background(), options, app.CartWorkflow, state)
+    if err != nil {
+        log.Fatalln("unable to execute workflow", err)
+    }
+
+    update := app.AddToCartSignal{Route: app.RouteTypes.ADD_TO_CART, Item: app.CartItem{ProductId:0, Quantity: 1}}
+    err = c.SignalWorkflow(context.Background(), workflowID, "", "ADD_TO_CART_CHANNEL", update)
+
+    resp, err := c.QueryWorkflow(context.Background(), workflowID, "", "getCart")
+    if err != nil {
+        log.Fatalln("Unable to query workflow", err)
+    }
+    var result interface{}
+    if err := resp.Get(&result); err != nil {
+        log.Fatalln("Unable to decode query result", err)
+    }
+    // Prints a message similar to:
+    // 2021/03/31 15:43:54 Received query result Result map[Email: Items:[map[ProductId:0 Quantity:1]]]
+    log.Println("Received query result", "Result", result)
+}
+```
+
+You now have a completed Temporal app.
+However, what you don't yet have are any tests, or a way for users to actually interface with this app.
+You'll complete those in the rest of this tutorial.
+
+## Write Workflow tests
+
+Temporal Workflows also make your code easier to test. Temporal provides testing utilities that help you stub out external services and programmatically advance time, which lets you unit test your Workflows. For an example, take a look at how [Descript uses Temporal](https://temporal.io/case-studies/descript-case-study).
+Temporal's testing environment also provides utilities for mocking Activities and testing logic that executes after a delay. That makes it less complicated to unit test Workflows that depend on external services or Workflows that involve long timeouts.
+
+To learn how to implement tests for this application, you can refer to [workflow_test.go](https://github.com/temporalio/temporal-ecommerce/blob/main/workflow_test.go) and [Notes on Testing](https://github.com/temporalio/temporal-ecommerce?tab=readme-ov-file#notes-on-testing) in the Github repo for this tutorial, or watch this walkthrough video:
+
+import { ResponsivePlayer } from '@site/src/components'
+
+<ResponsivePlayer url='https://www.youtube.com/watch?v=-GKxFDQSlEU' />
```

Comment:
Is the info in this link still relevant? Just checking as it's 3 years old

Reply:
Yup! This is actually doing testing for this specific tutorial, I deliberately kept it relevant.